tests/test_plugin_loader_git.py

```
@@ -0,0 +1,923 @@
+            handle.write("import missingdep_pipx\n\nclass Plugin:\n    pass\n")
+
+        orig_pipx_home = os.environ.get("PIPX_HOME")
+        orig_pipx_local_venvs = os.environ.get("PIPX_LOCAL_VENVS")
+        orig_virtual_env = os.environ.get("VIRTUAL_ENV")
+        orig_missingdep = sys.modules.get("missingdep_pipx")
+
+        for var in ("PIPX_HOME", "PIPX_LOCAL_VENVS", "VIRTUAL_ENV"):
+            os.environ.pop(var, None)
+        os.environ["PIPX_HOME"] = os.path.join(self.temp_plugins_dir, "pipx-home")
+
+        def _run_side_effect(cmd, *args, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_run_side_effect`

(ANN202)